main.js

```diff
@@ -0,0 +1,99 @@
+// let gc = document.querySelector('#gc');
+
+// gc.addEventListener('click', (e) =>
+// console.log(e.target.id +' was clicked'))
+
+let items = document.querySelectorAll('.card')
+
+let hasFlippedCard = false;
+let lockBoard = false;
+let firstCard, secondCard;
+let matched = ''
+
+
+function flipCard() {
+    if (lockBoard) return;
+    if (this === firstCard) return;
+    this.classList.add('flip');
+
+if (!hasFlippedCard) {
+    hasFlippedCard = true;
+    firstCard = this;
+    return;
+}
+secondCard=this;
+
+
+checkForMatch()
+}
+function checkForMatch() {
+    let isMatch = firstCard.dataset.framework === secondCard.dataset.framework;
+    isMatch ? disableCards() : unflipCards();
+}
+
+function disableCards() {
+    firstCard.removeEventListener('click', flipCard);
+    secondCard.removeEventListener('click', flipCard);
+    resetBoard();
+    console.log (matched += '1')
+    alert('Correct!')
+}
+
+(function win(){
+    if (matched === '111111') {
+        alert('Congrats You won!!!')
+    }
+}) ();
```

Comment:
I don't think this will ever run as an IIFE (that's the syntax you're using here with the function declaration wrapped in parens and followed by `()` -- an immediately invoked function expression. It's going to run right away. I think you _can_ check for a win using this strategy of tracking it -- maybe a counter could work here, and calling a function to check if all the cards are matched and therefore the game is over/won whenever you check a match? There are other ways but that's just one that comes to mind.